checks.d/f5.py

```diff
@@ -0,0 +1,474 @@
+                                version=3,                          # Default version is SNMP v3, other versions are insecure
+                                remote_port=port,                   # Use the specified port, defaulting to 161
+                                timeout=2,                          # Define the number of seconds before a timeout to be 2
+                                retries=3,                          # Define the maximum number of retries to be 3
+                                security_level="auth_with_privacy", # Use authentication with privacy
+                                security_username=user,             # Use the username from the instance
+                                privacy_protocol="AES",             # Define the privacy protocol to be AES
+                                privacy_password=privPassword,      # Use the privacy password from the instance
+                                auth_protocol="SHA",                # Define the authentication protocol to be SHA
+                                auth_password=authPassword          # Use the authentication password from the instance
+                            )                                       #
+            return session                                          # Otherwise, return the established snmp session with the F5
+        except Exception as e:                                      # If there are any exceptions
+            self.log.error("Could not establish session")           # log_level:error
+            self.log.error(e)                                       # log_level:error
+            sys.exit(1)                                             # Do not continue
+
+    def get_snmp_walk(self, session, OID):
+        """Perform an SNMP Walk operation on an object and return a dictionary of the results"""
+        try:
+            d = {}                                                      # Initialize a dictionary
+            lst = session.walk(str(OID))                                # Use the session to perform a walk
+            if len(lst) < 1:                                            # If no items were returned
+                self.log.warning("No items were returned while walking OID: %s", str(OID))
+            for item in lst:                                            # Iterate over the items
+                key = item.oid_index                                    # Group them by OID Index
+                if key not in d:                                        # If the key is not yet in the dictionary
+                    d[key] = []                                         # Set it's value to be an empty list
+                d[key].append((item.oid, item.value, item.snmp_type))   # And append the first instance tuple (OID, Value, SNMPType)
+                self.log.debug("Item found, appending to dictionary:\n\t%s", str(item))
```

Comment:
These logs will only be generated when the agent is run in debug mode, so a normal run should still be very quiet. If this is still an unwanted behavior, they can be removed.